backends/graphics/opengl/opengl-graphics.cpp

```diff
@@ -1297,36 +1297,30 @@ bool OpenGLGraphicsManager::saveScreenshot(const Common::String &filename) const
 	const uint width  = _windowWidth;
 	const uint height = _windowHeight;
 
-	// A line of a BMP image must have a size divisible by 4.
-	// We calculate the padding bytes needed here.
-	// Since we use a 3 byte per pixel mode, we can use width % 4 here, since
-	// it is equal to 4 - (width * 3) % 4. (4 - (width * Bpp) % 4, is the
-	// usual way of computing the padding bytes required).
-	// GL_PACK_ALIGNMENT is 4, so this line padding is required for PNG too
-	const uint linePaddingSize = width % 4;
-	const uint lineSize        = width * 3 + linePaddingSize;
-
 	Common::DumpFile out;
 	if (!out.open(filename)) {
 		return false;
 	}
 
-	Common::Array<uint8> pixels;
-	pixels.resize(lineSize * height);
-	GL_CALL(glReadPixels(0, 0, width, height, GL_RGB, GL_UNSIGNED_BYTE, &pixels.front()));
-
 #ifdef SCUMM_LITTLE_ENDIAN
 	const Graphics::PixelFormat format(3, 8, 8, 8, 0, 0, 8, 16, 0);
 #else
 	const Graphics::PixelFormat format(3, 8, 8, 8, 0, 16, 8, 0, 0);
 #endif
 	Graphics::Surface data;
-	data.init(width, height, lineSize, &pixels.front(), format);
+	data.create(width, height, format);
+	for (uint y = 0; y < height; ++y) {
+		GL_CALL(glReadPixels(0, height - y - 1, width, 1, GL_RGB, GL_UNSIGNED_BYTE, data.getBasePtr(0, y)));
```

Comment:
`glReadPixels` usually generates a VRAM -> RAM transfer. Having one such transfer per line does not seem to be very good for performance. Perhaps it would be better to have a function that vertically flips a surface in-place on the CPU if we want to remove that responsibility from the image codecs.

Reply:
OK, done.